spec/models/whitehall_asset_spec.rb

```diff
@@ -44,6 +44,17 @@
           expect(asset).not_to be_valid
           expect(asset.errors[:legacy_url_path]).to include('is already taken')
         end
+
+        context 'but the existing asset has been deleted' do
+          before do
+            asset.legacy_url_path = existing_asset.legacy_url_path
+            existing_asset.delete
+          end
+
+          it 'is valid' do
```

Comment:
Minor: It might be worth using the example name to explain things further, e.g. `it 'is valid because legacy_url_path is unique within the assets not marked as deleted' do`.

Reply:
I've updated this and force-pushed.